src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogPromotionActionType.php

```diff
@@ -80,12 +90,25 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                     'label' => false,
                 ]);
             })
+            ->addEventListener(FormEvents::SUBMIT, function(FormEvent $event): void {
+                /** @var CatalogPromotionActionInterface|null $data */
+                $data = $event->getData();
+                if ($data === null) {
+                    return;
+                }
+
+                $form = $event->getForm();
+
+                $actionConfigurationType = $this->actionConfigurationTypes[$data->getType()];
+                $form->add('configuration', $actionConfigurationType, [
```

Comment:
Why is it needed to be added on submit?

Reply:
without adding it to sumit form resets and we dont have inputs for selected  action type